mongo/with_transactions_test.go

```diff
@@ -259,6 +260,41 @@ func TestConvenientTransactions(t *testing.T) {
 		assert.True(t, ok, "expected context for abortTransaction to contain ctxKey")
 		assert.Equal(t, "foobar", ctxValue, "expected value for ctxKey to be 'world', got %s", ctxValue)
 	})
+	t.Run("commitTransaction timeout allows abortTransaction", func(t *testing.T) {
+		coll := db.Collection("test")
+
+		session, err := client.StartSession()
+		assert.Nil(t, err, "StartSession error: %v", err)
+		defer session.EndSession(context.Background())
+
+		err = WithSession(context.Background(), session, func(sessionContext SessionContext) error {
+			// Start transaction
+			err = session.StartTransaction()
+			assert.Nil(t, err, "StartTransaction error: %v", err)
+
+			// Insert a document
+			_, err := coll.InsertOne(sessionContext, bson.D{
+				{"ID", primitive.NewObjectID()},
+				{"val", 17},
+			})
+			assert.Nil(t, err, "InsertOne error: %v", err)
+
+			// Set a timeout of 0 for commitTransaction
+			commitTimeoutCtx, commitCancel := context.WithTimeout(sessionContext, 0)
+			defer commitCancel()
+
+			// commitTransaction results in context.DeadlineExceeded
+			commitErr := session.CommitTransaction(commitTimeoutCtx)
```

Comment:
Can you add a check to ensure that this does not have a `UnknownTransactionCommitResult` label?

Per some offline discussion, there might be more implications to this change.

If `CommitTransaction` currently adds an `UnknownTransactionCommitResult` error label when given a timed out context, the [commit loop](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/session.go#L209-L232) retry in `WithTransaction` may repeatedly attempt to commit a transaction until the two minute deadline expires.

I don't see a clear way to add that as a test. I think it would require timing out the context after. But if this change ensures that no `UnknownTransactionCommitResult` label is added, that might resolve that supposed issue.

cc @divjotarora if you'd like to weigh in. (I also wonder if there is a similar concern in CSOT if `timeoutMS` expires in the middle of the commit loop).

Reply:
Discussed offline again. `CommitTransaction()` does in fact add an `UnknownTransactionCommitResult` error label when given a timed out context. Then, the [CommitLoop](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/session.go#L209-L232) will repeatedly attempt to commit a transaction until the two minute (by default) deadline expires.

This does not seem like ideal behavior, so I've added a check on L212 in `WithTransaction()` to exit early in the event of a timeout.